baybe/simulation/lookup.py

```diff
@@ -33,31 +35,30 @@ def look_up_targets(
     Note that this does not create a new dataframe but modifies ``queries`` in-place.
 
     Args:
-        queries: A dataframe containing points to be queried.
-        targets: The targets whose values to look up.
         lookup: The lookup mechanism.
             See :func:`baybe.simulation.scenarios.simulate_scenarios` for details.
+        queries: A dataframe containing points to be queried.
+        targets: The targets whose values to look up.
         impute_mode: The used impute mode.
             See :func:`baybe.simulation.scenarios.simulate_scenarios` for details.
     """
-    # If no lookup is provided, invent some fake results
-    if lookup is None:
-        add_fake_results(queries, targets)
 
-    # Compute the target values via a callable
-    elif isinstance(lookup, Callable):
-        _look_up_targets_from_callable(queries, targets, lookup)
 
-    # Get results via dataframe lookup (works only for exact matches)
-    # IMPROVE: Although it's not too important for a simulation, this
-    #  could also be implemented for approximate matches
-    elif isinstance(lookup, pd.DataFrame):
-        _look_up_targets_from_dataframe(queries, targets, lookup, impute_mode)
+@look_up_targets.register
+def _look_up_fake_targets(
+    _: None, queries: pd.DataFrame, targets: Collection[Target]
```

Comment:
I do not know if this is proper coding practice, but I would prefer if the de-facto `lookup` argument was still called `lookup`

Reply:
This is outdated, since I tried to go through stuff commit by commit